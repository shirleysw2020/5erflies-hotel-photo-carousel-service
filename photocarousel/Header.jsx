// import React from 'react';
import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import { ThemeProvider } from 'styled-components';
// import Star from '../photos/star.png';
// import Superhost from '../photos/superhost.png';
// import Heart from '../photos/heart.png';
// import Share from '../photos/share.png';

const Header = ({list}) => (

  <Wrapper>
      <ThemeProvider theme={theme}>
          <TitleBlock>{list.listingTitle}</TitleBlock>
          <InfoData>
            <Logo src="https://5erflies.s3-us-west-1.amazonaws.com/icons/star.png"/>
            <RatingSpan>{list.rating}</RatingSpan>
            <SubColor>({list.numOfReviews})</SubColor>
            <Dot><span> • </span></Dot>
            {list.superHost === true ? <span><Logo host src="https://5erflies.s3-us-west-1.amazonaws.com/icons/superhost.png"/><SubColor>Superhost<Dot> • </Dot></SubColor></span> : null}
            <Underline location>{list.location}</Underline>
          </InfoData>
          <Socialmedia>
            <ShareButton>
              <Logo share src="https://5erflies.s3-us-west-1.amazonaws.com/icons/heart.png"/>
              <Underline dark>Share</Underline>
            </ShareButton>
            <ShareButton>
              <Logo share src="https://5erflies.s3-us-west-1.amazonaws.com/icons/share.png"/>
              {/* @styled-icons/boxicons-regular/Heart */}
              <Underline dark>Save</Underline>
            </ShareButton>
          </Socialmedia>
      </ThemeProvider>
  </Wrapper>
);


const theme = {
  titleColor: '#000',
  subColor: '#6e6d6d',
  padding: '4em'
}

const Wrapper = styled.div`
  /* margin-top: 40px */
  margin-bottom: 20px;
`;

const InfoData = styled.div`
  display: inline-block;
  font-size: 15px;
  padding-top: 20px;
  color: ${props => props.theme.color};
  font-family: Circular, Roboto, "Helvetica Neue", sans-serif;
`;

const Socialmedia = styled.div`
  display: inline-block;
  float: right;
  margin-top: 13px;
`;

const PaddingInner = styled.div`
  padding-right: 0.25rem;
`;

const TitleBlock = styled.section`
  width: 100%;
  margin: 0 auto;
  font-family: Circular, Roboto, "Helvetica Neue", sans-serif;
  color: ${props => props.theme.titleColor};
  font-size: 26px;
`;

const RatingSpan = styled.span`
  color: ${props => props.theme.titleColor};
  padding: 0em 0.15em;
`;

const SubColor = styled.span`
  color: ${props => props.theme.subColor};
  padding: 0em 0.15em;
`;

const Underline = styled.span`
  text-decoration: underline;
  color: ${props => props.dark ? 'black' : props.theme.subColor};
  padding: 0.2em 0.35em;
  float: right;
  cursor: pointer;
`;

const Dot = styled.span`
  padding: 0em 0.50em;
  color: ${props => props.theme.subColor};
`
const Logo = styled.img`
  width: ${props => props.host ? '10px' : '17px'};
  height: 16px;
  padding-right: 7px;
  padding-top: ${props => props.share ? '3px' : 'none'};
  float: ${props => props.share ? 'right' : 'none'};
  cursor: pointer;
`;

const ShareButton = styled.button`
  float: right;
  border: none;
  background-color: white;
  padding: 5px 6px;
  cursor: pointer;
  &: hover {
    background-color: #f2f4f5;
    padding: 5px 6px;
    border: none;
    border-radius: 8px;
    -webkit-transition: all 0.2s ease;
    transition: all 0.2s ease;
  }
`;

export default Header;