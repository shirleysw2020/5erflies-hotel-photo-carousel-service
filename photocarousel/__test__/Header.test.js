import React from 'react';
import styled from 'styled-components';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Header from '../Header.jsx';

const list = {
        "photos" : [
                "https://5erflies.s3-us-west-1.amazonaws.com/CalanteLuna/249fd66d-2b01-47e1-9c40-ca7f78887dd3.jpg",
                "https://5erflies.s3-us-west-1.amazonaws.com/CalanteLuna/004d5a66-ccf2-4b03-9bcf-93d3d735c6d5.jpg",
                "https://5erflies.s3-us-west-1.amazonaws.com/CalanteLuna/88f6875c-8e12-4265-9e78-0f5b43f37d57.jpg",
                "https://5erflies.s3-us-west-1.amazonaws.com/CalanteLuna/622f87e1_original.jpg",
                "https://5erflies.s3-us-west-1.amazonaws.com/CalanteLuna/900afa63-66d2-46e4-9101-fac8a8100938.jpg",
                "https://5erflies.s3-us-west-1.amazonaws.com/CalanteLuna/948a8be4_original.jpg"
        ],
        "ID" : 20,
        "listingTitle" : "CalanteLuna Relais - M'Illumino d'Immenso",
        "rating" : 4.01,
        "numOfReviews" : 473,
        "superHost" : false,
        "location" : "Gottliebchester Borders Italy",
        "__v" : 0
};

describe('<Header> components', () => {
  it('should have the Header component', () => {
    const wrapper = shallow(<Header list={list}/>)
    expect(wrapper.find('Logo')).toBeDefined();
  });

  it('should match snapshot created', () => {
    const tree = renderer.create(<Header list={list}/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});




// it ('should have subColor', () => {
//   const wrapper = mount(<Header />);
//   expect(wrapper.find('InfoData')).toHaveStyleRule('color', theme.subColor)
// });

// describe('Label', () => {
//   it('should render label and text', () => {
//     const wrapper = shallow(<Label required>Some Text</Label>).dive();

//     expect(wrapper.text()).toBe('Some Text');
//   });
// });

// const Logo = styled.img`
//   width: ${props => props.host ? '16px' : '17px'};
//   height: ${props => props.host ? '16px' : 'none'};
//   padding-bottom: 20px;
//   padding-right: 2px;
// `;