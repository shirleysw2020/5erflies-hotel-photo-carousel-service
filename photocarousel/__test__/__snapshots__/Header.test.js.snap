// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header> components should match snapshot created 1`] = `
<div
  className="sc-bdVaJa PeosY"
>
  <section
    className="sc-ifAKCX dmvJkz"
  >
    CalanteLuna Relais - M'Illumino d'Immenso
  </section>
  <div
    className="sc-bwzfXH kZXRah"
  >
    <img
      className="sc-dnqmqq bFJaqh"
      src="test-file-stub"
    />
    <span
      className="sc-EHOje bsuoWl"
    >
      4.01
    </span>
    <span
      className="sc-bZQynM BuzJs"
    >
      (
      473
      )
    </span>
    <span
      className="sc-htoDjs etCUlw"
    >
      <span>
         • 
      </span>
    </span>
    <img
      className="sc-dnqmqq kTZvYA"
      src="test-file-stub"
    />
    <span
      className="sc-bZQynM BuzJs"
    >
      <span>
        Superhost
      </span>
    </span>
    <span
      className="sc-htoDjs etCUlw"
    >
      <span>
         • 
      </span>
    </span>
    <span
      className="sc-gzVnrw Ajofz"
    >
      Gottliebchester Borders Italy
    </span>
  </div>
  <div
    className="sc-htpNat fTeqPD"
  >
    <button
      className="sc-iwsKbI hRMDpm"
    >
      <img
        className="sc-dnqmqq gStQyt"
        src="test-file-stub"
      />
      <span
        className="sc-gzVnrw lnKKkD"
      >
        Share
      </span>
    </button>
    <button
      className="sc-iwsKbI hRMDpm"
    >
      <img
        className="sc-dnqmqq gStQyt"
        src="test-file-stub"
      />
      <span
        className="sc-gzVnrw lnKKkD"
      >
        Save
      </span>
    </button>
  </div>
</div>
`;
